src/components/panel/panel.tsx

```diff
@@ -387,12 +387,13 @@ export class Panel
   }
 
   renderHeaderActionsEnd(): VNode {
-    const { close, hasEndActions, messages, closable } = this;
+    const { close, hasEndActions, messages, closable, hasMenuItems } = this;
     const text = messages.close;
 
     const closableNode = closable ? (
       <calcite-action
         aria-label={text}
+        data-test="close"
```

Comment:
Can you create a follow-up issue to look into having these data attributes removed from production builds? I think we may be able to do that by using Stencil's `Build.isTesting` flag.

Reply:
Can I just strip it out instead?

Reply:
Might be a stencil bug, but I tried `data-test={Build.isTesting ? "close" : undefined}` and the test failed.

Reply:
https://github.com/Esri/calcite-components/issues/6760